ORK1Kit/ORK1Kit/Common/ORK1Step.h

```diff
@@ -247,6 +247,15 @@ The default value of this property is 'nil'.
  */
 @property (nonatomic, copy, nullable) NSDictionary *userInfo;
 
+/**
+ When YES, upon loading this step, the ORK1TaskViewController will instruct
+ the delegate to save the task results.
```

Comment:
Forgot to mention some notes about this documentation:

- ResearchKit itself doesn't have any intrinsic concept of "saving task results" in the way that you mean here; and existing mentions of "save" elsewhere in ORK1TaskViewController's documentation tends to refer to updating the ORK1TaskResult with each step, or persisting restorationData, rather than handling the final survey result after completion.
- There is no new delegate method specific to saveOnArrival, so the "will instruct the delegate" phrasing is inaccurate.

The documentation for the `didFinishWithReason` delegate method states:

>  In most circumstances, the receiver should dismiss the task view controller in response to this method, and may also need to collect and process the results of the task.

So perhaps the `saveOnArrival` documentation should read more like that, and emphasize that the app is intended to instead perform that "collect and process" behavior upon reaching the step, e.g. in `stepViewControllerWillAppear`, and only perform dismissal or error handling behaviors in `didFinishWithReason`.

And could also mention that if the user arrives at any step with `saveOnArrival`, it will be the final step presented to the user.

Reply:
Good point - have made https://github.com/CareEvolution/ResearchKit/pull/131 to address.